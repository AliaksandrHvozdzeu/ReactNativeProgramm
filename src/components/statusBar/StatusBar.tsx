import React from 'react';
import {View} from 'react-native';
import {styles} from './styles';

type StatusBarProps = {};

const StatusBar: React.FC<StatusBarProps> = () => (
  <View style={styles.layout}>
    <View style={styles.statusBar} />
  </View>
);

export default StatusBar;
