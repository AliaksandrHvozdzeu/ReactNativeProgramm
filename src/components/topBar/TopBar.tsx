import React from 'react';
import {View} from 'react-native';
import StatusBar from '../statusBar';
import {styles} from './styles';

const TopBar = () => {
  return (
    <View style={styles.layout}>
      <StatusBar />
    </View>
  );
};

export default TopBar;
