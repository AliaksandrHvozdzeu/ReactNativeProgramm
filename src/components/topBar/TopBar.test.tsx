import 'react-native';
import React from 'react';
import {test, describe, expect} from '@jest/globals';
import TopBar from './TopBar';
import StatusBar from '../StatusBar';
import {render} from '@testing-library/react-native';

describe('TopBar tests', () => {
  test('Renders the TopBar correctly', () => {
    const tree = render(<TopBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('TopBar render works correctly', () => {
    render(<TopBar />);
  });

  test('Component TopBar presented in component', () => {
    const wrapper = render(<TopBar />);
    const testInstance = wrapper.root;
    expect(testInstance.findByType(StatusBar)).toBeTruthy();
  });
});
