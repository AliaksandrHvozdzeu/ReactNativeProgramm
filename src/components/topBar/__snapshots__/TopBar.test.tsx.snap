// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar tests Renders the TopBar correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "stretch",
        "height": 20,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#1C1F27",
          "height": 20,
          "width": "auto",
        }
      }
    />
  </View>
</View>
`;
